README.md

```diff
@@ -1,5 +1,57 @@
+## What aospy does
+This package provides tools for automating computations involving gridded climate data and the management of the results of those computations.  This enables the user to perform in parallel such commonly needed tasks as:
+
+* repeating a calculation across multiple simulations in a single climate model
+* repeating a calculation across the same simulations performed in multiple models, even if variable names or other quantities differ among the models or simulations
+* repeating a calculation across multiple timespans, both in terms of the start date and end date and in terms of sub-annual sampling: e.g. annual mean, seasonal-means, monthly means.
+* Computing multiple statistical  (e.g. mean, standard deviation) and physical (e.g. column integrals or averages; zonal integrals or averages) reductions on any given computation, both on a gridpoint-by-gridpoint basis and over an arbitrary number of geographical regions
+* any combination of the above, plus many more!
+
+In fact, aospy enables the user to specify an arbitrary combination of models, simulations, variables to compute, time ranges, months of the year, input data types, output data types, and other parameters and to submit them all for execution with a single command.  The desired computations are then executed in parallel, and their outputs are stored as netCDF files in a highly structured directory format with metadata imbued within (1) the netCDF files themselves, (2) the filenames, and (3) the directory structure.  This prevents the all-to-common occurrence of computations becoming that get saved to disk becoming obsolete as time passes (or for other users), due to insufficient metadata describing precisely what they are and how they were computed.
+
+## Design philosophy
+Key to enabling this automation and handling of model- and simulation-level idiosyncrasies is separating
+
+1. Code that describes the data that you want to work with
+2. Code that specifies any physical calculations you eventually want to perform
+3. Code that specifies the set of computations the user wishes to perform at a given time
+
+For (1), the user defines objects at three distinct levels: `Proj`, `Model`, and `Run`, that specify where the data is located that you want to work with.  For (2), the user defines `Var` objects that describe the physical quantities to be computed, including any functions that transform one or more directly model-outputted quantities into the ultimately desired quantity.  Once these objects have been defined, the user can proceed with (3) via a simple script that specifies any models, simulations, physical quantities, etc. to be performed.  aospy then permutes over the specified combinations of inputs and desired outputs, submits all the calculations in parallel, and then saves the output as netCDF files in a highly structured and metadata-infused directory and filename structure.
```

Comment:
The last sentence of this subsection:

> aospy then permutes over the specified combinations of inputs and desired outputs, submits all the calculations in parallel, and then saves the output as netCDF files in a highly structured and metadata-infused directory and filename structure.

is very similar to:

>In fact, aospy enables the user to specify an arbitrary combination of models, simulations, variables to compute, time ranges, months of the year, input data types, output data types, and other parameters and to submit them all for execution with a single command. The desired computations are then executed in parallel, and their outputs are stored as netCDF files in a highly structured directory format with metadata imbued within (1) the netCDF files themselves, (2) the filenames, and (3) the directory structure.

in the previous subsection.  I tend to like the prior subsection's version more.  Maybe consider removing the sentence here?